File: SuperMovie_server/src/api/Statistics/GetReleasedFilmTypeBor.cs
namespace SuperMovie.Server.Api.Statistics;

using WebSocketSharp;
using WebSocketSharp.Server;

public struct GetReleasedFilmTypeBorReq
{
}

public struct FilmTypeBoxOfficeRsp
{
    public string FilmType;
    public long FilmBoxOffice;
}

public struct GetReleasedFilmTypeBorRsp
{
    public List<FilmTypeBoxOfficeRsp> Collection;
}

//api : get_released_film_type_bor
public class GetReleasedFilmTypeBor : WebSocketBehavior
{
}

File: movieui/src/scripts/ws/Statistics/getReleasedFilmTypeBor.ts
import {wsUrlRoot} from "@/scripts/ws/meta"
import {createWebSocket, recvMsg, reqStringify, rspParse, sendMsg} from "@/scripts/ws/helper"

export {getReleasedFilmTypeBor}
export type {
    FilmTypeBoxOfficeRsp,
    GetReleasedFilmTypeBorReq,
    GetReleasedFilmTypeBorRsp
}

type GetReleasedFilmTypeBorReq =
    {}

type FilmTypeBoxOfficeRsp =
    {
        FilmType: string
        FilmBoxOffice: number
    }

type GetReleasedFilmTypeBorRsp =
    {
        Collection: FilmTypeBoxOfficeRsp[]
    }

async function getReleasedFilmTypeBor(req: GetReleasedFilmTypeBorReq) {
    const conn = createWebSocket(`${wsUrlRoot}/get_released_film_type_bor`)

    const task = recvMsg(conn)
    const reqJson = reqStringify(req)
    console.log('get_released_film_type_bor req:' + reqJson)
    console.log(reqJson)
    sendMsg(conn, reqJson).then()
    const msg = await task
    console.log('get_released_film_type_bor req:' + msg)

    return <GetReleasedFilmTypeBorRsp>rspParse(msg)
}